scikeras/wrappers.py

```diff
@@ -712,13 +712,30 @@ def fit(self, X, y, sample_weight=None) -> "BaseWrapper":
         sample_weight : array-like of shape (n_samples,), default=None
             Array of weights that are assigned to individual samples.
             If not provided, then each sample is given unit weight.
+        **kwargs : Dict[str, Any]
+            Extra arguments to route to ``Model.fit``.
+            This paremeter will be removed in a future release of SciKeras,
+            instead set fit arguments at initialization
+            (e.g., ``BaseWrapper(epochs=10, ...)``)
 
         Returns
         -------
         BaseWrapper
             A reference to the instance that can be chain called
             (ex: instance.fit(X,y).transform(X) )
         """
+        for k, v in kwargs.items():
+            warnings.warn(
+                "``kwargs`` will be removed in a future release of SciKeras."
+                f"Instead, set fit arguments at initialization (i.e., ``BaseWrapper({k}={v})``)"
+            )
+        self.set_params(
+            **{
+                (k if k.startswith("fit__") else "fit__" + k): v
+                for k, v in kwargs.items()
+            }
+        )
```

Comment:
@stsievert  what are your thoughts on these implementations? The main con I see is that subsequent calls will re-use the previous parameters. That is:

```python
est.fit(..., batch_size=32)
...
est.fit(...)  # implicitly uses batch_size=32 with no warnings to the user
```

Reply:
:-1: on this implementation. As a user, I would expect the params to behave the same unless I change them. Let the user do that, and issue an appropriate warning (like is being done).

[PEP 20][1]: "Explicit is better than implicit."

[1]:https://www.python.org/dev/peps/pep-0020/#the-zen-of-python

Reply:
Which part of the implementation are you 👎 on? This is basically what you proposed, plus the prefixing of `fit__` and `predict__`. Adding this prefix makes it behave more like the old implement, in the sense that passing a kwarg to `fit` won't "set" it for `predict` and visa versa.

Overall, I think there are 3 options:
1) We don't support `**kwargs` at all and _force_ users to make the change.
2) This implementation.
3) This implementation + we "unset"/"revert" the kwargs at the end. I haven't tested implementing this, and I do think it would add some complexity, but in theory it should be possible.

Reply:
`wrappers.py` diff cleaned up. (3) is looking pretty good now.